avocado/utils/software_manager.py

```diff
@@ -517,6 +536,72 @@ def provides(self, name):
         else:
             return None
 
+    def build_dep(self, name):
+        """
+        Install build-dependencies for package [name]
+
+        :param name: name of the package
+
+        :return True: If build dependencies are installed properly
+        """
+        option = '-y --tolerant'
+        cmd = 'yum-builddep %s %s' % (option, name)
+
+        try:
+            process.system(cmd, sudo=True)
+            return True
+        except process.CmdError, details:
+            log.error(details)
+            return False
+
+    def prepare_source(self, name, dest_path):
+        """
+        Downloads the source package and prepares it in the given dest_path
+        to be ready to build.
+
+        :param name: name of the package
+        :param dest_path: destination_path
+
+        :return final_dir: path of ready-to-build directory
+        """
+        path = tempfile.mkdtemp(prefix='avocado_software_manager')
+        src_path = os.environ['HOME']
+        build_option = "-bp"
+        final_dir = None
+        if dest_path is not None:
+            build_option += " --define '_builddir %s'" % dest_path
+        else:
+            log.error("Please provide a valid path")
+            return ""
+        pkg_list = ['yum-utils', 'rpm-build']
+        for pkg in pkg_list:
+            if not self.check_installed(pkg):
+                if not self.install(pkg):
+                    log.error("SoftwareManager (YumBackend) can't get packages"
+                              "with dependency resolution: Package 'yum-utils'"
+                              "could not be installed")
+                    return ""
+        try:
+            process.system_output('yumdownloader --source %s --destdir %s' % (name, path))
+            print [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
+            src_rpm = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))][0]
```

Comment:
Well, it is a tmpfile, but still it looks really ugly to depend on the first file present in the tmpdir. You can get the name from the `process.system_output`, can't you?

Reply:
@ldoktor Actually tried that, but looks like system_output does not catch the last line (probably because of the progress bar) , which is why I opted for this method. Any better suggestion would be helpful. Thanks.

Reply:
@ldoktor I have found another way to do it as the current implementation looks quite ugly. Please let know if this helps.